File: 1.WEB_MES/backend/server-standard/src/main/kotlin/com/gmin/serverstandard/application/domain/model/Correspondent.kt
package com.gmin.serverstandard.application.domain.model

import java.time.LocalDateTime

/**
 * 거래처 도메인 모델
 *
 * 거래처의 기본 정보를 담고 있는 도메인 엔티티입니다.
 */
data class Correspondent(
    val correspondentId: Int,
    val name: String,
    val type: String,
    val ceo: String,
    val businessNumber: String,
    val phoneNumber: String,
    val email: String,
    val address: String,
    val detailAddress: String,
    val note: String,
    val correspondentPhotoUri: String,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
) {
    companion object {
        // 거래처 타입 상수
        const val TYPE_PURCHASE = "매입"
        const val TYPE_SALE = "매출"
        const val TYPE_BOTH = "매출/매입"
        
        // 거래처 타입 유효성 검사
        fun isValidCorrespondentType(type: String): Boolean {
            return type == TYPE_PURCHASE || 
                   type == TYPE_SALE || 
                   type == TYPE_BOTH
        }
    }
}

File: 1.WEB_MES/frontend/MESALL.Shared/Models/Correspondent.cs
using MESALL.Shared.Enums;

namespace MESALL.Shared.Models;

// 거래처 모델 클래스
public class Correspondent
{
    public int CorrespondentId { get; set; }
    public string Name { get; set; } = string.Empty;
    public CorrespondentType Type { get; set; }
    public string Ceo { get; set; } = string.Empty;
    public string BusinessNumber { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string DetailAddress { get; set; } = string.Empty;
    public string Note { get; set; } = string.Empty;
    public string CorrespondentPhotoUri { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

// 거래처 생성 요청 모델
public class CreateCorrespondentRequest
{
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Ceo { get; set; } = string.Empty;
    public string BusinessNumber { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string DetailAddress { get; set; } = string.Empty;
    public string Note { get; set; } = string.Empty;
    public string CorrespondentPhotoUri { get; set; } = string.Empty;
}

// 거래처 수정 요청 모델
public class UpdateCorrespondentRequest : CreateCorrespondentRequest
{
}